src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -78,7 +79,28 @@ private struct CustomDebugInformationRow
         private readonly List<CustomDebugInformationRow> _customDebugInformationTable = new List<CustomDebugInformationRow>();
 
         private readonly Dictionary<DebugSourceDocument, int> _documentIndex = new Dictionary<DebugSourceDocument, int>();
-        private readonly Dictionary<IImportScope, int> _scopeIndex = new Dictionary<IImportScope, int>();
+        private readonly Dictionary<IImportScope, int> _scopeIndex = new Dictionary<IImportScope, int>(ImportScopeEqualityComparer.Instance);
+
+        /// <summary>
+        /// Import scopes are associated with binders (in C#) and thus multiple instances might be created for a single set of imports.
+        /// We consider scopes with the same parent and the same imports the same.
+        /// Internal for testing.
+        /// </summary>
+        internal sealed class ImportScopeEqualityComparer : IEqualityComparer<IImportScope>
+        {
+            public static readonly ImportScopeEqualityComparer Instance = new ImportScopeEqualityComparer();
+
+            public bool Equals(IImportScope x, IImportScope y)
+            {
+                return (object)x == y ||
+                       x != null && y != null && Equals(x.Parent, y.Parent) && x.GetUsedNamespaces().SequenceEqual(y.GetUsedNamespaces());
+            }
+
+            public int GetHashCode(IImportScope obj)
+            {
+                return Hash.CombineValues(obj.GetUsedNamespaces(), obj.Parent?.GetHashCode() ?? 0);
```

Comment:
I think we need to change the name of the method(s) so that it would hard to make similar mistake in the future.